bfa-automation-java/src/test/java/com/bfa/core/page/GooglePageAutomationTest.java

```diff
@@ -0,0 +1,76 @@
+package com.bfa.core.page;
+
+import com.bfa.core.BrowserType;
+import org.junit.After;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Created by ninhdoan on 12/28/16.
+ */
+public class GooglePageAutomationTest {
+
+   private static final String SELENIUM_DRIVER_KEYWORD = "Selenium WebDriver";
+
+   private static final String SELENIUM_HOME_URL = "http://www.seleniumhq.org/projects/webdriver/";
+
+   private static final String GOOGLE_HOME_URL = "https://www.google.com";
+
+   private GooglePage googlePage;
+
+   @After
+   public void closeBrowser() {
+      googlePage.close();
+   }
+
+   private void openGoogleInFirefox() {
+      googlePage = GooglePage.openGoogle(BrowserType.FIREFOX);
+   }
+
+   private void openGoogleInChrome() {
+      googlePage = GooglePage.openGoogle(BrowserType.CHROME);
+   }
+
+   @Test
+   public void shouldGetGoogleHomePageWhenOpenGoogleInFirefox() {
+      openGoogleInFirefox();
+
+      verifyCurrentUrlIsGoogleHome();
+   }
+
+   @Test
+   public void shouldGetGoogleHomePageWhenOpenGoogleInChrome() {
+      openGoogleInChrome();
+
+      verifyCurrentUrlIsGoogleHome();
+   }
+
+   private void verifyCurrentUrlIsGoogleHome() {
+      String currentUrl = googlePage.waitToGetNewUrl();
+
+      assertThat(currentUrl).contains(GOOGLE_HOME_URL);
+   }
+
+   @Test
+   public void shouldGoToSeleniumHomePageWhenClickTheFirstResultAfterGoogleSeleniumDriverInFirefox() {
+      openGoogleInFirefox();
+
+      shouldGoToSeleniumHomePageWhenClickFirstResultAfterGoogleSeleniumDriver();
+   }
+
+   @Test
+   public void shouldGoToSeleniumHomePageWhenClickTheFirstResultAfterGoogleSeleniumDriverInChrome() {
+      openGoogleInChrome();
+
+      shouldGoToSeleniumHomePageWhenClickFirstResultAfterGoogleSeleniumDriver();
+   }
+
+   private void shouldGoToSeleniumHomePageWhenClickFirstResultAfterGoogleSeleniumDriver() {
+      googlePage.search(SELENIUM_DRIVER_KEYWORD).openSearchResultAt(1);
+
+      String currentUrl = googlePage.waitToGetNewUrl();
+
+      assertThat(currentUrl).isEqualTo(SELENIUM_HOME_URL);
+   }
+}
```

Comment:
We will be using BDD as testing approach, the design as unit test here is not suitable. Please revert to the old one.